state-synchronizer/src/executor_proxy.rs

```diff
@@ -170,37 +158,28 @@ impl ExecutorProxyTrait for ExecutorProxy {
             verified_target_li,
             intermediate_end_of_epoch_li,
         )?;
-        self.publish_on_chain_config_updates(reconfig_events).await
+        self.publish_on_chain_config_updates(reconfig_events)
     }
 
-    async fn get_chunk(
+    fn get_chunk(
         &self,
         known_version: u64,
         limit: u64,
         target_version: u64,
     ) -> Result<TransactionListWithProof> {
-        self.storage_read_client
+        self.storage
             .get_transactions(known_version + 1, limit, target_version, false)
-            .await
     }
 
-    async fn get_epoch_proof(
-        &self,
-        start_epoch: u64,
-        end_epoch: u64,
-    ) -> Result<ValidatorChangeProof> {
+    fn get_epoch_proof(&self, start_epoch: u64, end_epoch: u64) -> Result<ValidatorChangeProof> {
         let validator_change_proof = self
-            .storage_read_client
-            .get_epoch_change_ledger_infos(start_epoch, end_epoch)
-            .await?;
+            .storage
+            .get_epoch_change_ledger_infos(start_epoch, end_epoch)?;
         Ok(validator_change_proof)
     }
 
-    async fn get_ledger_info(&self, version: u64) -> Result<LedgerInfoWithSignatures> {
-        let (_, _, li_chain, _) = self
-            .storage_read_client
-            .update_to_latest_ledger(version, vec![])
-            .await?;
+    fn get_ledger_info(&self, version: u64) -> Result<LedgerInfoWithSignatures> {
+        let (_, _, li_chain, _) = self.storage.update_to_latest_ledger(version, vec![])?;
```

Comment:
we should really call `self.storage.get_latest_ledger_info()` here and avoid adding `.update_to_latest_ledger()` to `trait DbReader` at this point.

Reply:
Here it is not to get the latest_ledger_info but the first after this waypoint.

Reply:
Good catch! Looks like we need to expose `get_epoch(version)` and use it combined with `get_epoch_change_ledger_infos()` you've already introduced above.

Reply:
I add a new api called `get_ledger_info`